src/dotnetTestExplorer.ts

```diff
@@ -111,21 +111,33 @@ export class DotnetTestExplorer implements TreeDataProvider<TestNode> {
         }
     }
 
+    private enableBuild(): string {
+        const dotnetBuildOption = Utility.getConfiguration().get<boolean>("build");
+        return dotnetBuildOption ? "--no-build" : "";
+    }
+
+    private enableRestore(): string {
+        const dotnetRestoreOption = Utility.getConfiguration().get<boolean>("restore");
+        return dotnetRestoreOption ? "--no-restore" : "";
+    }
+
     private loadTestStrings(): Thenable<string[]> {
         this.evaluateTestDirectory();
 
         let msBuildRootTestMsg = Utility.getConfiguration().get<string>("msbuildRootTestMsg");
         msBuildRootTestMsg = msBuildRootTestMsg ? msBuildRootTestMsg : "The following Tests are available:";
 
+        const command = `dotnet test -t -v=q ${this.enableBuild()} ${this.enableRestore()}`;

```

Comment:
Will investigate that and tune it accordingly in this upcoming commit. I'm fixing some mistakes I overlooked.